pkg/cmd/repo/default/default.go

```diff
@@ -0,0 +1,250 @@
+package base
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"sort"
+	"strings"
+
+	ctx "context"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/context"
+	"github.com/cli/cli/v2/git"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+type iprompter interface {
+	Select(string, string, []string) (int, error)
+}
+
+type DefaultOptions struct {
+	IO         *iostreams.IOStreams
+	Remotes    func() (context.Remotes, error)
+	HttpClient func() (*http.Client, error)
+	Prompter   iprompter
+
+	Repo     ghrepo.Interface
+	ViewMode bool
+}
+
+func NewCmdDefault(f *cmdutil.Factory, runF func(*DefaultOptions) error) *cobra.Command {
+	opts := &DefaultOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		Remotes:    f.Remotes,
+		Prompter:   f.Prompter,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "default [<repository>]",
+		Short: "Configure default repository",
+		Long: heredoc.Doc(`
+			Set default repository for current directory.
+
+			gh uses the default repository for things like:
+
+			 - viewing, creating, and setting the default base for  pull requests
+			 - viewing and creating issues
+			 - viewing and creating releases
+			 - working with Actions
+			 - adding secrets
```

Comment:
Since there are multiple types of secrets
```suggestion
			 - adding repository and environment secrets
```